### PR TITLE
Fixed readFy

### PR DESCRIPTION
**Pull request by Vivian-liyw**
- fixed readFy
- may need implement testing on frontend
- changed error codes

### CODE REVIEW

File: controllers/fyController.js [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+        orderby = orderby + ' ,' + order;
+      } else {
+        orderby = ' ORDER BY' + order;
+      }
+    }
+  })
+
+  //calculate the number of pages
+  const queryCount = "SELECT COUNT(*) AS count FROM students";
+  var pages = 0;
```
Comment (KurtXiaoZ) [inline controllers/fyController.js:282]: pages is not updated when the user changes conditions.
image

File: controllers/userController.js [hunk 2]
```diff
@@ -305,35 +304,21 @@ exports.readUser = async (req, res) => {
...
-      return res.send({ status: STATUS_CODE.SUCCESS });
-    }
-  } catch (error) {
-    return res.send({ status: STATUS_CODE.ERROR, result: error });
-  }
-  return res.send({ status: STATUS_CODE.SUCCESS});
-}
+// return empty list when no value is found in DB
+// return the max Visions group number
+exports.visionsNums = async (req, res) => {
```
Comment (KurtXiaoZ) [inline controllers/userController.js:309]: Doesn't respond with the correct list of vision groups for the Visions Assignment page

Comment (KurtXiaoZ) [reply controllers/userController.js:309]: In addition, we need another API similar to visionsNums for the filter of VUceptor
image

**Review by KurtXiaoZ — CHANGES_REQUESTED** [review]

File: controllers/fyController.js [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+      return res.send({ status: STATUS_CODE.SUCCESS });
+    }
+  } catch (error) {
+    return res.send({ status: STATUS_CODE.ERROR, result: error });
+  }
+  return res.send({ status: STATUS_CODE.SUCCESS});
+};
+
+//get all first year students, return a json object
+exports.readFy = async (req, res) => {
```
Comment (KurtXiaoZ) [inline controllers/fyController.js:165]: When I change row_start, 'pages' is returned with a value 0.
image

**Review by KurtXiaoZ — CHANGES_REQUESTED** [review]

File: controllers/fyController.js (hunk 1 above)
Comment (Vivian-liyw) [reply controllers/fyController.js]: great point!
Fixed the bug in userController and fyController

**Review by KurtXiaoZ — CHANGES_REQUESTED** [review]

**Review by KurtXiaoZ — APPROVED** [review]
lgtm! Great work

**Pull request closed without merging**